### ISSUE TITLE
World currently unbeatable

### ISSUE THREAD

**Issue by sflavelle** [sflavelle/APmanual_blastcorps#1]
After rebasing the world to Unstable 20240528 and writing up some (newbie-ish) rules for Blast Corps' level tiering, with the goal set to `shuttle_clear` (the earliest possible finish), Archipelago currently believes the world is unbeatable.

```
Could not access required locations for accessibility check. Missing: [Ironstone Mine - Scientist, Ebony Coast - Scientist, Tempest City - Scientist, Oyster Harbor - Scientist, Glory Crossing - Scientist, Victory - Shuttle Clear, FINISH!]
Uncaught exception
Traceback (most recent call last):
  File "/home/lily/Archipelago/APsource/Generate.py", line 565, in <module>
    multiworld = main()
                 ^^^^^^
  File "/home/lily/Archipelago/APsource/Generate.py", line 239, in main
    return callback(erargs, seed)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lily/Archipelago/APsource/Main.py", line 414, in main
    raise Exception("Game appears as unbeatable. Aborting.")
Exception: Game appears as unbeatable. Aborting.
```

An initial read through the location/region list and `requires` don't show anything out of the ordinary, so it's probably one of [the new rules I've written up.](https://github.com/sflavelle/APmanual_blastcorps/blob/405e48616ddf3c668c26d6b17ab25115255c9c46/hooks/Rules.py#L10-L87).

**Comment by FuzzyGamesOn** [comment]
At a glance, noticed locsChecked isn't defined here, so this requirement function will always fail: https://github.com/sflavelle/APmanual_blastcorps/blob/main/hooks/Rules.py#L83

**Comment by sflavelle** [comment]
As noted in Discord, the `rankPoints` function is half-baked at the moment but intends to give some logic to the 30 in-game rank locations in Ranksanity.

With my current test yaml though, this function should be unused, as those locations are disabled:

```yaml
Manual_BlastCorps_Splatsune:
  goal: shuttle_clear
  all_levels: true
  time_attack: false
  ranksanity: false
```

**Comment by sflavelle** [comment]
Fixed with [commit 7cbeaa5](https://github.com/sflavelle/APmanual_blastcorps/commit/7cbeaa5b1db1c9d73fc6cdafa600259b06f18e8f) - discovered I should not be using state.locations_reached for access calculations. Swapped over to can_reach(), tested that items were placed as expected. See the commit description for more details.
